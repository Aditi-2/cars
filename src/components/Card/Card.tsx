import styled from "@emotion/styled";

export interface CardProps {
  manufacturer?: string;
  model?: string;
  vehicle?: string;
}

export const Card: React.FC<CardProps> = ({manufacturer, model, vehicle}) => {
  return (
    <CardStyle>
      {manufacturer && <p className="card-details">Manufacturer: <span>{manufacturer}</span></p>}
      {model && <p className="card-details">Model: <span>{model}</span></p>}
      {vehicle && <p className="card-details">Vehicle: <span>{vehicle}</span></p>}
    </CardStyle>
  );
};

const CardStyle = styled.div`
  padding: 1rem;
  margin: 1rem auto;
  border: 1px solid #dfdfdf;
  border-radius: 2px;
  text-align: left;
  width: 85%;
  @media only screen and (min-width: 1200px) {
    width: 30%;
  }
  & p {
    color: #7d7d7d;
    & span {
      color: #000000;
    }
  }
`;
