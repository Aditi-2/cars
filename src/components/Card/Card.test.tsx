import React from "react";
import { create } from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { Card, CardProps } from "./Card";

const baseProps: CardProps = {
  manufacturer: 'A',
  model: 'B',
  vehicle: 'C'
};

describe("Card Test", () => {
  it("Should match the snapshot", () => {
    const component = create(<Card {...baseProps} />);
    expect(component).toMatchSnapshot();
  });

  it("Should render Card", () => {
    render(<Card {...baseProps} />);
    const text = screen.getByText('A')
    expect(text).toBeInTheDocument();
  });
});
