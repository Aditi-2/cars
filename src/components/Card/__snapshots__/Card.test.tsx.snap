// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test Should match the snapshot 1`] = `
<div
  className="css-1w6rvfd"
>
  <p
    className="card-details"
  >
    Manufacturer: 
    <span>
      A
    </span>
  </p>
  <p
    className="card-details"
  >
    Model: 
    <span>
      B
    </span>
  </p>
  <p
    className="card-details"
  >
    Vehicle: 
    <span>
      C
    </span>
  </p>
</div>
`;
